evadb/executor/create_function_executor.py

```diff
@@ -394,13 +416,26 @@ def handle_forecasting_function(self):
         ]
         if len(existing_model_files) == 0:
             logger.info("Training, please wait...")
+            for column in data.columns:
+                if column != "ds" and column != "unique_id":
+                    data[column] = data.apply(
+                        lambda x: self.convert_to_numeric(x[column]), axis=1
+                    )
             if library == "neuralforecast":
                 model.fit(df=data, val_size=horizon)
+                model.save(model_path, overwrite=True)
             else:
+                # The following lines of code helps eliminate the math error encountered in statsforecast when only one datapoint is available in a time series
+                for col in data["unique_id"].unique():
+                    if len(data[data["unique_id"] == col]) == 1:
+                        data = data._append(
+                            [data[data["unique_id"] == col]], ignore_index=True
+                        )
+
```

Comment:
We need to be explicit. If some data points are discarded, the user needs to be aware.  Can we print out some warning information in this case?

Reply:
We are not discarding any data point here. The math domain error occurs when there is only a single data point; this code helps replicate that value instead of throwing the error. Technically, it is not possible to forecast with just one data point.